### DISCUSSION TITLE
Full arguments support for Protobuf MCAP writer in Python

### DISCUSSION THREAD

**Discussion by IvDmNe** [foxglove/mcap#1108] [General]
In Python Base MCAP Writer class supports many arguments in it's  [__init__](https://github.com/foxglove/mcap/blob/main/python/mcap/mcap/writer.py#L54) method.
But a Protobuf MCAP writer has only output stream argument. So, now If I want to have MCAP file with LZ4 compression, I need to get it from protobuf mcap writer with ZSTD compression and then use CLI to convert the file to LZ4. Also, chunking options are not available.

I think it would be nice to  be able to pass all arguments of default Writer also to Protobuf Writer. Is this feature planned to be added?

If no, and If you think it's worth to be added, I can make a few changes and create pull request.

Possible solutions:
 - make similar arguments of base writer to Protobuf writer and pass them in __init__ function;
 - make Base Writer a parent class of Protobuf writer (Were there any reasons that protobuf writer was created as a wrapper of base Writer instead of using inheritance?)

**Comment by defunctzombie** [comment]
@james-rms something to ticket for 1.0 libraries project

**Comment by james-rms** [comment]
Yep - this makes sense, thanks for the suggestion.
